jenkins/ci.suse.de/pipelines/openstack-ardana-gerrit.Jenkinsfile

```diff
@@ -16,21 +16,53 @@ pipeline {
 
   stages {
 
-    stage('validate commit message') {
-      steps {
-        script {
-          currentBuild.displayName = "#${BUILD_NUMBER}: ${gerrit_change_ids}"
+    stage('static analysis') {
+      parallel {
+        stage('validate commit message') {
+          steps {
+            script {
+              currentBuild.displayName = "#${BUILD_NUMBER}: ${gerrit_change_ids}"
+            }
+            cleanWs()
+            sh '''
+              git clone $git_automation_repo --branch $git_automation_branch automation-git
+              export LC_ALL=C.UTF-8
+              export LANG=C.UTF-8
+
+              source /opt/gitlint/bin/activate
+
+              echo $GERRIT_CHANGE_COMMIT_MESSAGE | base64 --decode | gitlint -C automation-git/scripts/jenkins/gitlint.ini
+            '''
+          }
         }
-        cleanWs()
-        sh '''
-          git clone $git_automation_repo --branch $git_automation_branch automation-git
-          export LC_ALL=C.UTF-8
-          export LANG=C.UTF-8
 
-          source /opt/gitlint/bin/activate
+        stage('ansible syntax check') {
+          steps {
+            sh '''
+              set -x
 
-          echo $GERRIT_CHANGE_COMMIT_MESSAGE | base64 --decode | gitlint -C automation-git/scripts/jenkins/gitlint.ini
-        '''
+              # Make sure we are running in a gerrit context
+              if [ -n "$GERRIT_PROJECT" ]; then
+                remote="https://$GERRIT_HOST/$GERRIT_PROJECT"
+                if ! [ -d "$GERRIT_PROJECT" ]; then
+                  git clone $remote $GERRIT_PROJECT
+                fi
+                cd $GERRIT_PROJECT
+
+                git fetch $remote $GERRIT_PATCHSET_REVISION
+                git reset --hard FETCH_HEAD
+                git clean -f
+
+                if [ -d tests ] && [ -r tests/test.yml ] ; then
```

Comment:
Where is this `test` folder supposed to come from ? I checked a few Gerrit repositories and I couldn't find it there.

Reply:
You're right, I couldn't find any either. Which lead me to discover I've been copying the wrong syntax job from gozer. I'm going to move the syntax check portion to its own pull request.